testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,693 @@
+        ignores = fileToSet(p, "ignores");
+    }
+
+    private static Set<String> fileToSet(Properties p, String name) {
+        String fn = p.getProperty(name);
+        Set<String> wl = new HashSet<>();
+        if (fn == null) {
+            return wl;
+        } else {
+            try (BufferedReader br = new BufferedReader(new FileReader(fn))) {
+                    for(String line; (line = br.readLine()) != null; ) {
+                        // Support for comments (line starts with #)
+                        // Essentially only add if it does not start with #
+                        if (!line.startsWith("#")) {
+                            wl.add(line);
+                        }
+                    }
+                    return Collections.unmodifiableSet(wl);
+                } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    public StateCapture(String testName) {
+        this.testName = testName;
+        try {
+            dBuilder = DocumentBuilderFactory.newInstance()
+                .newDocumentBuilder();
+        } catch (ParserConfigurationException ex) {
+            ex.printStackTrace();
+        }
+        setup();
+    }
+
+    // Constructor to control if it's a class-level state capture
+    public StateCapture(String testName, boolean classLevel) {
+        this(testName);
+        this.classLevel = classLevel;
+    }
+
+    /**
+     * Call this method to capture the state, save it, and do the diffing if all required
+     * states for the current test have been captured.
+     * @throws IOException 
+     */
+    public void runCapture() {
+        // if we already captured 4 states for the current test, then we are ready to diff
+        if (currentTestStates.size() == MAX_NUM) {
+            if (enableLogging) {
+                logger.startTimer();
+            }
+            if (enableLogging) {
+                logger.stopTimeAndUpdate(StateCaptureLogger.Task.DIFF);
+                logger.saveToFileAndReset();
+            }
+        }
+    }
+
+    Set<String> File2SetString(String path) {
+        File file = new File(path);
+        Set<String> keys = new HashSet<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                keys.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    public void setup() {
+        subxmlFold = Configuration.config().getProperty("replay.subxmlFold");
+        rootFold = Configuration.config().getProperty("replay.rootFold");
+        diffFold = Configuration.config().getProperty("replay.diffFold");
+        slug = Configuration.config().getProperty("replay.slug");
+        reflectionFile = Configuration.config().getProperty("replay.reflectionFold") + "/reflection.txt";
```

Comment:
If all that is needed from the reflectionFold is the file, can just have the property point to the file directly instead of constructing the path with hard-coded file names.